ecs/pod_creator.go

```diff
@@ -27,11 +30,134 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+	var out cocoa.ECSPod
+	var err error
+
+	mergedPodOptions := cocoa.MergeECSPodCreationOptions(opts...)
+
+	translatedIn, secrets, err := m.TranslateRegisterTaskDefinitionInput(ctx, mergedPodOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, translatedIn)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	taskDef := cocoa.ECSTaskDefinition{
+		ID:    registerOut.TaskDefinition.TaskDefinitionArn,
+		Owned: utility.TruePtr(),
+	}
+
+	runOut, err := m.client.RunTask(ctx, &ecs.RunTaskInput{
+		Cluster:        cocoa.MergeECSPodExecutionOptions(mergedPodOptions.ExecutionOpts).Cluster,
+		TaskDefinition: taskDef.ID,
+	})
+	if err != nil || runOut.Failures != nil || runOut.Tasks == nil {
+		return nil, errors.Wrap(err, "running task")
+	}
+
+	out = &BasicECSPod{
+		client: m.client,
+		vault:  m.vault,
+		resources: cocoa.ECSPodResources{
+			TaskID:         runOut.Tasks[0].TaskArn,
```

Comment:
You have to check both `len(runOut.Tasks) == 1` or it'll panic. Also `runOut.Tasks[0].TaskArn != nil` or else you won't have a task ARN.